packages/vscode-extension/src/webview/views/PaywallView.tsx

```diff
@@ -0,0 +1,166 @@
+import { useEffect, useState } from "react";
+import "./PaywallView.css";
+import { PricePreviewResponse } from "@paddle/paddle-js";
+import classNames from "classnames";
+import Button from "../components/shared/Button";
+import RadonBackgroundImage from "../components/RadonBackgroundImage";
+import usePaddle from "../hooks/usePaddle";
+
+type SubscriptionPlan = "monthly" | "yearly";
+
+// FIXME: Add production Paddle price IDs when ready
+const RadonIDEProMonthlyPriceID = window.RNIDE_isDev ? "pri_01k1g12g3y3tqvpzw8tcyrsd1y" : "";
+const RadonIDEProYearlyPriceID = window.RNIDE_isDev ? "pri_01k1g8d3h0mhbtr5hfd9e4n8yg" : "";
+
+const benefits = [
+  "All the Free features",
+  "Redux UI plugin",
+  "React Query plugin",
+  "CPU Profiler integration",
+  "React Profiler integration",
+  "React Scan integration",
+  "Radon AI assistant",
+  "Early access to new features",
+  "Priority support",
+];
+
+function PaywallView() {
+  const [selectedPlan, setSelectedPlan] = useState<SubscriptionPlan>("yearly");
+  const paddle = usePaddle();
+  const [prices, setPrices] = useState<PricePreviewResponse | null>(null);
+  const [isLoadingPrices, setIsLoadingPrices] = useState(false);
+  const [pricesError, setPricesError] = useState<string | null>(null);
+
+  async function getPrices() {
+    setIsLoadingPrices(true);
+    setPricesError(null);
+
+    try {
+      const response = await paddle?.PricePreview({
+        items: [
+          { quantity: 1, priceId: RadonIDEProMonthlyPriceID },
+          { quantity: 1, priceId: RadonIDEProYearlyPriceID },
+        ],
+        address: {
+          countryCode: "US",
+        },
+      });
+
+      if (!response) {
+        throw new Error("Failed to fetch prices from Paddle - no response received");
+      }
+
+      setPrices(response);
+      setIsLoadingPrices(false);
```

Comment:
I think that the line `setIsLoadingPrices(false)` in the `try` block might be redundant, as we already have this line in the finally block which will execute regardless of whether an error occurs